### PR TITLE
feat: Clicking the status bar item stops and starts the server

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

File: editors/code/src/ctx.ts [hunk 1]
```diff
@@ -204,13 +204,13 @@ export class Ctx {
...
 
-    setServerStatus(status: ServerStatusParams) {
+    setServerStatus(status: ServerStatusParams | { health: "stopped" }) {
         let icon = "";
         const statusBar = this.statusBar;
         switch (status.health) {
             case "ok":
-                statusBar.tooltip = status.message ?? "Ready";
-                statusBar.command = undefined;
+                statusBar.tooltip = (status.message ?? "Ready") + "Click to stop.";
```
Comment (bjorn3) [inline editors/code/src/ctx.ts:212]: There should probably be a space here.

Comment (Veykril) [reply editors/code/src/ctx.ts:212]: Saw that earlier as well, fixed in https://github.com/rust-lang/rust-analyzer/pull/13512

**Comment by lnicola** [comment]
https://user-images.githubusercontent.com/308347/198942818-1e2d8c47-0a5b-44ed-b0c6-9cb4807b00dc.mp4

**Pull request merged**